src/Command/Redis/XADD.php

```diff
@@ -35,7 +35,23 @@ public function setArguments(array $arguments)
         $args = [];
 
         $args[] = $arguments[0];
-        $options = $arguments[3] ?? [];
+
+        $optionsOffset = 3;
+        $idOffset = 2;
+        $pushRefArg = false;
+
+        $refArguments = ['KEEPREF', 'DELREF', 'ACKED'];
+        $stringArgs = array_filter($arguments, function ($arg) {
+            return is_string($arg);
+        });
+
+        if (count(array_intersect($refArguments, $stringArgs)) == 1) {
```

Comment:
This whole block is overcomplicated. I think new argument could be another entry in `$options` something like `['trimming' => 'keepref']`.

This way we would follow the existing approach and we can much easier handle this by checking if key 'trimming' exists and if the value corresponds to one of the possible trimming types